src/core/math/curve.js

```diff
@@ -154,7 +156,7 @@ class Curve {
     clone() {
         /** @type {this} */
         const result = new this.constructor();
-        result.keys = extend(result.keys, this.keys);
+        result.keys = this.keys.map(key => [...key]);
```

Comment:
the extend function seemed to do a deep copy, but the new solution does a shallow copy only, which is probably not intended?

Reply:
The `keys` are an array of arrays of two numbers. So I can't see how this would go wrong.

Reply:
yep that should be fine , great!